### ISSUE TITLE
Expose public API to allow overflow when Modal is open

### ISSUE THREAD

**Issue by Aweary** [react-bootstrap/react-overlays#86]
Currently the `style` object has a hard coded `overflow` value of `'hidden'`.

https://github.com/react-bootstrap/react-overlays/blob/master/src/ModalManager.js#L68

 We have a use case where our UX team would like the user to be able to scroll the content behind the modal. We are not explicitly passing in a container, so it's defaulting to the body, and we'd like to avoid just styling the body element directly.

Would you guys accept a PR that implements an `allowScroll` prop? Or, if you want to generalize this, a `containerStyle` prop that is spread on the default `style` object?

**Comment by jquense** [comment]
hi. We'd gladly take a PR, personally I'd prefer `containerStyle` vs the less flexible `allowScroll`

**Comment by Aweary** [comment]
@jquense `dom-helpers/style` doesn't seem to support numerical values (e.g., `{width: 500}`) is that be choice? I was hoping to just spread the `containerStyle` prop into the `style` property that is passed to `css` in `ModalManager` but it would break with numerical values.

**Comment by Aweary** [comment]
@jquense would you guys accept a PR on `dom-helpers` that just defaults numerical values to pixels? So `500` would be parsed as `500px`?

**Comment by jquense** [comment]
@Aweary thats hard, becasue some properties don't have units, like `font-weight`, which react and jquery maintain blacklists (https://github.com/jquery/jquery/blob/master/src/css.js#L186), I'm not sure its worth it to have to maintain on such a minimal library, is it too much to specify container style have units? It might be worth leaving dom-helpers alone and require https://github.com/facebook/react/blob/master/src/renderers/dom/shared/CSSProperty.js directly to add units.

@taion thoughts?

**Comment by Aweary** [comment]
> thats hard, becasue some properties don't have units, like font-weight

Good point, that would defeat the purpose I was aiming for which is to make `dom-helpers/style` accept the same style properties that  React can support. It just seems potentially confusing to have a style prop on a React component that doesn't implement the style interface of React itself.

Using `CSSProperty. isUnitlessNumber` is an interesting idea, I'm just hesitant to import internal React utilities.

**Comment by Aweary** [comment]
Hey @jquense coming back to this, if you guys are OK with importing a module from React internals, using `CSSProperty` is a decent solution. No new dependencies and you'll always be in sync with React's behavior.

On the other hand, if you guys don't want to do that, maybe we can use something like https://github.com/rofrischmann/unitless-css-property

**Comment by taion** [comment]
I feel like that support should belong in dom-helpers – in which case it would be ideal to not have a React dependency there.

**Comment by Aweary** [comment]
Oh, right, dom-helpers doesn't depend on React (duh), so would you guys be open to using that unitless-css-property package in dom-helpers for this then?

**Comment by taion** [comment]
Not my package! @jquense?

**Comment by jquense** [comment]
👍  sounds good to me

**Comment by taion** [comment]
I believe @jquense added support for not handling container overflow in https://github.com/react-bootstrap/react-overlays/pull/113.

**Comment by emilebres** [comment]
Hi, I had the same problem (container style overflow set to hidden). 

I managed to solve it using the changes made in #113 but I had to jump through hoops:

- importing ModalManager from 'react-overlays/lib/ModalManager'
- setting the prop `manager` of `Modal` to `new ModalManager({ handleContainerOverflow: false })`

It works but it's kinda complicated and not documented AFAIK.
If you're interested, I would happy to add it to the docs or expose it as a prop for `Modal`.
